staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1/conversion.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/client-go/pkg/apis/clientauthentication"
+)
+
+func Convert_clientauthentication_ExecCredentialSpec_To_v1alpha1_ExecCredentialSpec(in *clientauthentication.ExecCredentialSpec, out *ExecCredentialSpec, s conversion.Scope) error {
```

Comment:
curious why a manual conversion was required here

Reply:
Not sure if there is a way to avoid it based on the type structure:

alpha:

```go
// ExecCredenitalSpec holds request and runtime specific information provided by
// the transport.
type ExecCredentialSpec struct {
	// Response is populated when the transport encounters HTTP status codes, such as 401,
	// suggesting previous credentials were invalid.
	// +optional
	Response *Response `json:"response,omitempty"`

	// Interactive is true when the transport detects the command is being called from an
	// interactive prompt.
	// +optional
	Interactive bool `json:"interactive,omitempty"`
}
```

beta:

```go
// ExecCredentialSpec holds request and runtime specific information provided by
// the transport.
type ExecCredentialSpec struct {
	// Cluster contains information to allow an exec plugin to communicate
	// with the kubernetes cluster being authenticated to.
	Cluster Cluster `json:"cluster"`
}
```

internal:

```go
// ExecCredenitalSpec holds request and runtime specific information provided by
// the transport.
type ExecCredentialSpec struct {
	// Response is populated when the transport encounters HTTP status codes, such as 401,
	// suggesting previous credentials were invalid.
	// +optional
	Response *Response

	// Interactive is true when the transport detects the command is being called from an
	// interactive prompt.
	// +optional
	Interactive bool

	// Cluster contains information to allow an exec plugin to communicate
	// with the kubernetes cluster being authenticated to.
	Cluster Cluster
}
```

Reply:
I thought missing fields generated a warning but did not require a custom Convert method... do they not?

Reply:
If I remove the manually written custom Convert method and regen everything, the resulting code expects the functions to be present but does not generate them, leading to compile failure:

```
# k8s.io/client-go/pkg/apis/clientauthentication/v1beta1
../../../../../pkg/apis/clientauthentication/v1beta1/zz_generated.conversion.go:65:10: undefined: Convert_clientauthentication_ExecCredentialSpec_To_v1beta1_ExecCredentialSpec
../../../../../pkg/apis/clientauthentication/v1beta1/zz_generated.conversion.go:126:12: undefined: Convert_clientauthentication_ExecCredentialSpec_To_v1beta1_ExecCredentialSpec
# k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1
../../../../../pkg/apis/clientauthentication/v1alpha1/zz_generated.conversion.go:55:10: undefined: Convert_clientauthentication_ExecCredentialSpec_To_v1alpha1_ExecCredentialSpec
../../../../../pkg/apis/clientauthentication/v1alpha1/zz_generated.conversion.go:96:12: undefined: Convert_clientauthentication_ExecCredentialSpec_To_v1alpha1_ExecCredentialSpec
FAIL	k8s.io/client-go/plugin/pkg/client/auth/exec [build failed]
```

Reply:
hmm, ok